### PR TITLE
Cleanup aad in beginning of build

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
Performs the AAD cleanup of test applications at the start of the build

## Related issues
Addresses the role conflict exception that occasionally occurs in builds:
image

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [x] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

**Review by Ivanidzo4ka — APPROVED** [review]
:shipit:

File: build/pr-pipeline.yml [hunk 1]
```diff
@@ -83,6 +83,8 @@ stages:
           catch
           {}
           New-AzResourceGroup -Name "$(ResourceGroupName)" -Location "$(ResourceGroupRegion)" -Force
+  - template: ./jobs/cleanup-aad.yml
```
Comment (Ivanidzo4ka) [inline build/pr-pipeline.yml:86]: >cleanup [](start = 21, length = 7)

I find it's a bit strange to be part of this stage, and not one below (aadTestEnvironment)
In my head it feels more natural to bundle together work with AAD and put cleanup right before we setup test environment.

Comment (brendankowitz) [reply build/pr-pipeline.yml:86]: It needs to happen before the Template deployment. I'll see if it makes sense to move the setup step here too

Comment (LTA-Thinking) [reply build/pr-pipeline.yml:86]: Could you add it to the aadTestEnvironment stage as a second template before the add template? That way all the aad steps would be together.

Comment (brendankowitz) [reply build/pr-pipeline.yml:86]: I changed some of the dependencies so I now have this, I think this looks better:
image

Comment (Ivanidzo4ka) [reply build/pr-pipeline.yml:86]: >It needs to happen before the Template deployment.

Any chance you can elaborate? I'm not sure what Template you talking about.

Looking on that picture, why not put Setup AAD test as job into Provision Environment stage? (and give it better name than setup :D )

**Review by LTA-Thinking — APPROVED** [review]

File: build/pr-pipeline.yml [hunk 2]
```diff
@@ -83,20 +83,22 @@ stages:
...
           New-AzResourceGroup -Name "$(ResourceGroupName)" -Location "$(ResourceGroupRegion)" -Force
-
-- stage: aadTestEnvironment
-  displayName: Setup AAD Test Environment
-  dependsOn:
-  - provisionEnvironment
-  - DockerBuild
-  jobs:
-  - template: ./jobs/add-aad-test-environment.yml
+  - template: ./jobs/cleanup-aad.yml
```
Comment (LTA-Thinking) [inline build/pr-pipeline.yml:86]: The provision environment stage now needs to depend on the SemVer stage since add AAD uses the environment name.

**Review by LTA-Thinking — CHANGES_REQUESTED** [review]

File: build/pr-pipeline.yml (hunk 2 above)
Comment (brendankowitz) [reply build/pr-pipeline.yml:87]: Thanks! I think I've keep the provision stage separate and move the aad cleanup/create to its own stage

Comment (brendankowitz) [reply build/pr-pipeline.yml:94]: Dependencies now look like this. 
image

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**